### PR TITLE
Added NEP MLIP relax, static, and MD makers

### PR DESCRIPTION
**Pull request by naik-aakash**
## Summary: Added NEP MLIP relax, static, and MD maker

I have added makers so NEP potentials can also be used in atomate2 workflows. 
A [general-purpose NEP potential for 16 elemental metals and their alloys](https://arxiv.org/abs/2311.04732) was made available last year. 

Implemented Makers at this point use the ase-calculator implemented in the [calorine](https://gitlab.com/materials-modeling/calorine) package

## Todos: 

- [x] Add tests

### CODE REVIEW

**Comment by naik-aakash** [comment]
Hi @utf , this PR is ready for review and could be merged .

File: src/atomate2/forcefields/md.py [hunk 1]
```diff
@@ -333,6 +333,17 @@ def calculator(self) -> Calculator:
...
 
 
+@dataclass
+class NEPMDMaker(ForceFieldMDMaker):
+    """Perform an MD run with NEP."""
+
+    name: str = f"{MLFF.NEP} MD"
+    force_field_name: str = f"{MLFF.NEP}"
+    calculator_kwargs: dict = field(
+        default_factory=lambda: {"model_filename": "gap.xml"}
```
Comment (JaGeo) [inline src/atomate2/forcefields/md.py:343]:
```suggestion
        default_factory=lambda: {"model_filename": "nep.txt"}
```

Comment (naik-aakash) [reply src/atomate2/forcefields/md.py:343]: Oh thanks for the catch. Fixed it

**Review by JaGeo — APPROVED** [review]

File: tests/forcefields/test_jobs.py [hunk 2]
```diff
@@ -295,6 +297,79 @@ def test_gap_static_maker(si_structure: Structure, test_dir):
...
+        fix_symmetry=fix_symmetry,
+        calculator_kwargs={
+            "model_filename": test_dir / "forcefields" / "nep" / "nep.txt"
+        },
+    ).make(al2_au_structure)
+
+    # run the flow or job and ensure that it finished running successfully
+    responses = run_locally(job, ensure_success=True)
+
+    # validation the outputs of the job
```
Comment (QuantumChemist) [inline tests/forcefields/test_jobs.py:357]: I know it's just a comment 🙈 
```suggestion
    # validate the outputs of the job
```

**Comment by utf** [comment]
Looks great, thanks @naik-aakash.

**Pull request merged**